src/data_gradients/datasets/FolderProcessor.py

```diff
@@ -117,3 +116,107 @@ def __getitem__(self, index):
+        labels_dir: str,
+        config_path: str,
+        image_extensions: List[str],
+        label_extensions: List[str],
+        verbose: bool,
+    ) -> List[Tuple[str, str]]:
+        image_label_folder_iterator = ImageLabelFilesIterator(
+            images_dir=images_dir,
+            labels_dir=labels_dir,
+            image_extensions=image_extensions,
+            label_extensions=label_extensions,
+            verbose=verbose,
+        )
+        file_ids = self._load_file_ids(config_path=config_path)
+        filename_to_images_with_labels_files = {get_filename(image_path): (image_path, label_path) for (image_path, label_path) in image_label_folder_iterator}
+
+        images_with_labels_files = []
+        for file_id in file_ids:
+            if file_id in filename_to_images_with_labels_files:
+                images_with_labels_files.append(filename_to_images_with_labels_files[file_id])
+            elif verbose:
+                logger.warning(
+                    f"No file with `file_id={file_id}` found in `images_dir={images_dir}` and/or `labels_dir={labels_dir}`. "
+                    f"Hide this message by setting `verbose=False`."
+                )
+
+        if images_with_labels_files == []:
+            error_msg = (
+                f"Out of {len(file_ids)} file ids found in `config_path={config_path}`, "
+                f"no matching file found in `images_dir={images_dir}` and/or `labels_dir={labels_dir}`."
+            )
+            if not verbose:
+                error_msg += "\nSet `verbose=True` for more information."
+            raise RuntimeError(error_msg)
+        elif len(images_with_labels_files) != len(file_ids):
+            logger.warning(
+                f"Out of {len(file_ids)} file ids found in `config_path={config_path}`, "
+                f"{len(images_with_labels_files)} were found in both `images_dir={images_dir}` and `labels_dir={labels_dir}`. "
+                f"Hide this message by setting `verbose=False`."
+            )
+
+        return images_with_labels_files
+
+    def _load_file_ids(self, config_path: str) -> List[str]:
+        if not os.path.exists(config_path):
+            raise FileNotFoundError(f"The config file `{config_path}` does not exist.")
+
+        with open(config_path, "r") as f:
+            try:
+                file_ids = f.read().split()
+            except Exception as e:
+                raise e(f"Could not properly parse `config_path={config_path}`") from e
+
+        if file_ids == []:
+            raise RuntimeError(f"`config_path={config_path}` is empty and contains no file IDs.")
+
+        return file_ids
+
+    def __len__(self):
+        return len(self.images_with_labels_files)
+
+    def __getitem__(self, index):
+        return self.images_with_labels_files[index]
+
+    def __iter__(self):
+        for image_label_file in self.images_with_labels_files:
+            yield image_label_file
+
+
+def get_filename(file_name: str) -> str:
```

Comment:
Will this only be used in the context of this class ?
If so, please add it as a static method, otherwise lets find a better place for it.

Reply:
https://github.com/Deci-AI/data-gradients/pull/117/commits/833fab718d0e591e59c9a41701c7015f94a99876